### PR TITLE
CB-3073 add ddl resource, remove ddl controller

### PR DESCRIPTION
**Pull request by tfokina**

### CODE REVIEW

File: webapp/packages/plugin-ddl-viewer/src/DdlViewer/DDLViewerTabPanel.tsx [hunk 1]
```diff
@@ -20,38 +20,36 @@ import { SQLCodeEditorLoader } from '@cloudbeaver/plugin-sql-editor';
...
-        bindings={{
-          autoCursor: false,
-        }}
-        value={controller.metadata}
-        dialect={controller.dialect}
-        readonly
-      />
-      <MenuBar menu={menu} style={MENU_BAR_DEFAULT_STYLES} />
-    </wrapper>
+    <Loader state={[ddlResource, connectionDialectResource]}>
```
Comment (Wroud) [inline webapp/packages/plugin-ddl-viewer/src/DdlViewer/DDLViewerTabPanel.tsx]: Loader now can be skipped (Suspense)

File: webapp/packages/plugin-ddl-viewer/src/DdlViewer/DdlResource.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+
+import { injectable } from '@cloudbeaver/core-di';
+import { NavNodeInfoResource } from '@cloudbeaver/core-navigation-tree';
+import { CachedMapResource, ResourceKey, ResourceKeyUtils, isResourceAlias, GraphQLService } from '@cloudbeaver/core-sdk';
+
+@injectable()
+export class DdlResource extends CachedMapResource<string, string> {
```
Comment (Wroud) [inline webapp/packages/plugin-ddl-viewer/src/DdlViewer/DdlResource.ts:14]: great job!

File: webapp/packages/plugin-ddl-viewer/src/DdlViewer/DDLViewerTabPanel.tsx [hunk 3]
```diff
@@ -20,38 +20,36 @@ import { SQLCodeEditorLoader } from '@cloudbeaver/plugin-sql-editor';
...
-  });
-  // TODO: not triggered in switch case with lazy
-  // useTab(folderId, () => controller.load());
+  const connectionInfoResource = useService(ConnectionInfoResource);
+  const ddlResource = useResource(DDLViewerTabPanel, DdlResource, nodeId);
 
-  if (controller.isLoading) {
-    return <Loader />;
-  }
+  const connection = connectionInfoResource.getConnectionForNode(nodeId);
```
Comment (Wroud) [inline webapp/packages/plugin-ddl-viewer/src/DdlViewer/DDLViewerTabPanel.tsx]: probably it's better to load active project connections before
```
const connectionInfoResource = useResource(DDLViewerTabPanel, ConnectionInfoResource, ConnectionInfoActiveProjectKey);

const connection = connectionInfoResource.resource.getConnectionForNode(nodeId);
```

**Review by Wroud — CHANGES_REQUESTED** [review]

File: webapp/packages/plugin-ddl-viewer/src/DdlViewer/DDLViewerTabPanel.tsx (hunk 3 above)
Comment (tfokina) [reply webapp/packages/plugin-ddl-viewer/src/DdlViewer/DDLViewerTabPanel.tsx]: Done

File: webapp/packages/plugin-ddl-viewer/src/DdlViewer/DDLViewerTabPanel.tsx (hunk 1 above)
Comment (tfokina) [reply webapp/packages/plugin-ddl-viewer/src/DdlViewer/DDLViewerTabPanel.tsx]: Fixed

**Review by Wroud — APPROVED** [review]

**Pull request merged**